app/forms/work_form.rb

```diff
@@ -0,0 +1,95 @@
+# typed: false
+# frozen_string_literal: true
+
+require 'reform/form/coercion'
+
+class WorkForm < Reform::Form
+  feature Coercion
+
+  property :work_type
+  property :subtype, default: 'manuscript'
+  property :title
+  property :contact_email
+  property :abstract
+  property :citation
+  property :collection_id
+  property :license
+  property :agree_to_terms
+
+  property 'created(1i)', virtual: true
+  property 'created(2i)', virtual: true
+  property 'created(3i)', virtual: true
+  property 'created_range(1i)', virtual: true
+  property 'created_range(2i)', virtual: true
+  property 'created_range(3i)', virtual: true
+  property 'created_range(4i)', virtual: true
+  property 'created_range(5i)', virtual: true
+  property 'created_range(6i)', virtual: true
+  property 'published(1i)', virtual: true
+  property 'published(2i)', virtual: true
+  property 'published(3i)', virtual: true
+  property :creation_type, virtual: true, default: 'single'
+
+  def sync(*)
+    model.created_edtf = case creation_type
+                         when 'range'
+                           deserialize_edtf_range(:created_range)
+                         else
+                           deserialize_edtf(:created)
+                         end
+
+    model.published_edtf = deserialize_edtf(:published)
+
+    super
+  end
+
+  validates :title, presence: true
+
+  collection :contributors,
+             populator: lambda { |fragment:, **|
+               # find out if incoming Contributor is already added.
+               item = contributors.find_by(id: fragment['id']) if fragment['id'].present?
+
+               if fragment['_destroy'] == '1'
+                 contributors.delete(item)
+                 return skip!
+               elsif fragment['first_name'].blank?
```

Comment:
We used to reject if last name was blank and now we reject if first name is blank? Why the change?

Reply:
I don't really know what the requirements should be.  It was arbitrary the first time and it's arbitrary now.

Reply:
@jcoyne should we at least retain the arbitrariness for now? I know names are fraught with peril, so either way we are likely to get it wrong. :laughing: I'll add a planning discussion topic about this.

Reply:
https://github.com/sul-dlss/happy-heron/issues/232